### PR TITLE
infoschema: add TiFlash to `cluster_info` table

### PR DESCRIPTION
**Pull request by leiysky**
### What problem does this PR solve?

Problem Summary: TiFlash cannot be identified correctly in INFORMATION_SCHEMA.CLUSTER_INFO.

### What is changed and how it works?

What's Changed: Add a retriever to retrieve TiFlash stores.

How it Works: Check store label.

### Related changes
Nothing

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
Start a cluster with tiflash, and query its CLUSTER_INFO.

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### CODE REVIEW

**Comment by leiysky** [comment]
@crazycs520 PTAL

File: infoschema/tables.go [hunk 1]
```diff
@@ -1305,15 +1316,49 @@ func GetTiKVServerInfo(ctx sessionctx.Context) ([]ServerInfo, error) {
...
+				GitHash:        store.GitHash,
+				StartTimestamp: store.StartTimestamp,
+			})
+		}
+	}
+	return servers, nil
+}
+
+// GetTiFlashServerInfo returns all TiFlash nodes information of cluster
+func GetTiFlashServerInfo(ctx sessionctx.Context) ([]ServerInfo, error) {
```
Comment (lonng) [inline infoschema/tables.go:1335]: Seems we can eliminate the duplicated code?

Comment (leiysky) [reply infoschema/tables.go]: Fixed

File: infoschema/perfschema/tables.go [hunk 2]
```diff
@@ -265,7 +265,7 @@ func dataForRemoteProfile(ctx sessionctx.Context, nodeType, uri string, isGorout
 	)
 	switch nodeType {
 	case "tikv":
-		servers, err = infoschema.GetTiKVServerInfo(ctx)
+		servers, err = infoschema.GetStoreServerInfo(ctx)
```
Comment (djshow832) [inline infoschema/perfschema/tables.go:268]: Is the `nodeType` also `tikv` for TiFlash?

Comment (leiysky) [reply infoschema/perfschema/tables.go:268]: The `nodeType` is hard coded as `tikv` or `pd`, see: https://github.com/pingcap/tidb/blob/e2274a7790e982e4e717f6ec267af35e5f1cfedc/infoschema/perfschema/tables.go#L205-L220

Maybe we could handle it later.

**Pull request merged**